packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -132,13 +132,32 @@ export class JobService {
     const tokenFee = mul(fee, rate);
     const tokenTotalAmount = add(tokenFundAmount, tokenFee);
 
-    const { manifestUrl, manifestHash } = await this.saveManifest({
-      ...dto,
-      requestType,
-      submissionsRequired,
-      requesterDescription,
-      fundAmount: tokenFundAmount,
-    });
+    if (requestType == JobRequestType.FORTUNE) {
+      ({ manifestUrl, manifestHash } = await this.saveManifest({
+        ...(dto as JobFortuneDto),
+        requestType,
+        fundAmount: tokenFundAmount,
+      }));
+    } else {
+      dto = dto as JobImageLabelBinaryDto;
+      ({ manifestUrl, manifestHash } = await this.saveManifest({
+        data: {
+          data_url: dto.dataUrl,
+        },
+        annotation: {
+          labels: dto.labels.map((item) => ({ name: item })),
+          description: dto.requesterDescription,
+          type: requestType,
+          job_size: 10,
```

Comment:
Let's use env vars for these hardcoded values because we don't want to change the code every time we decide to change these values.